### PR TITLE
`HTTPClient`: debug log when performing redirects

### PR DESCRIPTION
**Pull request by NachoSoto**
This will be used to detect the load balancer redirecting to the load shedder. See also #2362.

### CODE REVIEW

File: Sources/Networking/HTTPClient/HTTPClient.swift [hunk 1]
```diff
@@ -336,13 +337,17 @@ private extension HTTPClient {
...
             self.handle(urlResponse: urlResponse,
                         request: request,
                         urlRequest: urlRequest,
                         data: data,
                         error: error)
         }
+
+        if #available(iOS 15.0, *) {
+            task.delegate = RedirectLoggerTaskDelegate()
+        }
```
Comment (aboedo) [inline Sources/Networking/HTTPClient/HTTPClient.swift:350]: we should always log - the load shedder happens automatically and isn't dependent on OS version. The only thing that is dependent on OS version is Offline Entitlements

**Review by aboedo — CHANGES_REQUESTED** [review]
ready to go after we remove the os check!

File: Sources/Networking/HTTPClient/HTTPClient.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Networking/HTTPClient/HTTPClient.swift:350]: `URLSessionTask.delegate` is iOS 15 only, but I'll do this using `URLSession.delegate` instead.

Comment (NachoSoto) [reply Sources/Networking/HTTPClient/HTTPClient.swift:350]: Oh wait, but that delegate doesn't have anything for following redirects...

**Review by aboedo — APPROVED** [review]

**Pull request merged**